### PR TITLE
CUDA sparse gemm multiply

### PR DESCRIPTION
**Pull request by cnpetra**

### CODE REVIEW

**Review by nychiang — APPROVED** [review]
Tried to check the usages from https://docs.nvidia.com/cuda/cusparse/index.html and some examples, and all the implementations LGTM.

**Comment by nychiang** [comment]
BTW, the pnnl CI fails due to the missing head files. For example:
.../src/LinAlg/hiopMatrixSparseCsrCuda.hpp:587:42: error: 'cusparseSpGEMMDescr_t' has not been declared

**Comment by cnpetra** [comment]
> /LinAlg/hiopMatrixSparseCsrCuda.hpp

looks like cusparseSpGEMM was added only recently (since cuda 11.0). @CameronRutherford what version of CUDA is on newell and marianas ?

File: src/Optimization/hiopKKTLinSysSparseCondensed.cpp [hunk 1]
```diff
@@ -270,12 +281,12 @@ bool hiopKKTLinSysCondensedSparse::build_kkt_matrix(const double& delta_wx_in,
...
 
-  SUM->print();
-  Hess_csr_->print();
+  //SUM->print();
+  //Hess_csr_->print();
   delete SUM;
   delete Hess_lower_csr_cuda;
   delete Hess_upper_csr_cuda;
-
+  delete JtDiagJ_cuda;
```
[the remark this answers is not in the document]
Comment (cnpetra) [reply src/Optimization/hiopKKTLinSysSparseCondensed.cpp:289]: the `delete` is already guarded

**Comment by cnpetra** [comment]
> We can update Marianas to use cuda < 11.4, but Newell is stuck with cuda < 11 due to it still using CentOS 7. Since we are the main users of Newell, it seems to be our responsibility to get that upgraded. Since upgrading the OS seems to be something that won't happen, I am proposing that we use Ascent for our Power9 cuda 11 tests. If this means disabling tests on Newell, then unfortunately that might be the only option.
> 
> Additionally, if this is upgrading the cuda requirement for all future versions of HiOp, we should certainly include this in a new release, and update HiOp's package.py to make everyone's life easier.

OK. That would do it. We have a CI at LLNL for CUDA code as well as.

**Pull request merged**